hollow/src/main/java/com/netflix/hollow/core/index/HollowPrefixIndex.java

```diff
@@ -85,9 +86,9 @@ public HollowPrefixIndex(HollowReadStateEngine readStateEngine, String type, Str
     @SuppressWarnings("WeakerAccess")
     public HollowPrefixIndex(HollowReadStateEngine readStateEngine, String type, String fieldPath,
             int estimatedMaxStringDuplicates) {
+        Objects.requireNonNull(readStateEngine, "Hollow Prefix Key Index creation failed because read state wasn't initialized");
```

Comment:
Did you intentionally flip the order here? (as compared to the other cases, not as compared to the previous code here) Might be good to keep all three consistent

Reply:
ACK, and added type to the message like the others